File: 0020-valid-parentheses/solution.java
class Solution {
    public boolean isValid(String str) {
        Stack<Character> st=new Stack<>();
        for(char ch:str.toCharArray()){
            if(ch=='('||ch=='['||ch=='{'){
                st.push(ch);
            }
            
            if(ch==')'||ch=='}'||ch==']'){
                if(st.isEmpty()) return false;
                char top=st.pop();
                if(ch==')'&&top!='(' ||
                ch==']'&&top!='[' ||
                ch=='}'&&top!='{'){
                    return false;
                }
            }
        } 
        return st.isEmpty();
    }
}

File: 0020-valid-parentheses/solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        list1=[]
        for i in s:
            if i=='(' or i=='[' or i=='{':
                list1.append(i)
            elif i==')' or i==']' or i=='}':
                if not list1:
                    return False
                x=list1.pop()
                if (x=='(' and i!=')') or (x=='[' and i!=']') or (x=='{' and i!='}'):
                        return False
        return len(list1)==0
